### DISCUSSION TITLE
ERROR: cache lookup failed for type xxxxx (SQLSTATE XX000)", err:(*pgconn.PgError)(0xc0003984b0)

### DISCUSSION THREAD

**Discussion by csphilli** [jackc/pgx#2015] [Q&A] [answered]
In my codebase, I haven't seen this error until I introduced an enum to my migrations. Strangely, the tests will pass just fine if run in isolation. But I use testify's suite to run all tests and this is when the error starts occurring.

The full error is
```
pgconn.PgError{Severity:"ERROR", Code:"XX000", Message:"cache lookup failed for type 36071", Detail:"", Hint:"", Position:0, InternalPosition:0, InternalQuery:"", Where:"unnamed portal parameter $4", SchemaName:"", TableName:"", ColumnName:"", DataTypeName:"", ConstraintName:"", File:"lsyscache.c", Line:2838, Routine:"getTypeInputInfo"}
```

What does that error indicate? A problem with the database? Something I'm not doing correctly with enums using pgx?

I found [this](https://github.com/sqlc-dev/sqlc/issues/2116#issuecomment-1493299852) relating to sqlc (which i'm also using) but the code pertains to registering types with pgx in a situation where the user needs to use CopyFrom. Since I'm not registering the types, and the tests pass in isolation, would this apply to my code? I've tried implementing it as well, but it didn't solve my problem.

This is my setup function:

```
func (suite *UsersUnitTestSuite) SetupTest() {
	c := config.LoadConfig()

	db, err := storage.InitTestPostgresAndMigrateUp()
	assert.Nil(suite.T(), err)
	suite.db = db

	dsn := storage.CreatePostgresDSN(c.PgDsnTemplate, c.PgHostname, c.PgUsername, c.PgPassword, c.PgDb)

	suite.ctx = CreateTestContext()

	connPool, err := pgxpool.New(suite.ctx, dsn)
	assert.Nil(suite.T(), err)

	assert.Nil(suite.T(), connPool.Ping(suite.ctx))
	suite.connPool = connPool

	InitDbStore(connPool)
}
```

I don't want to dump a lot of code yet into this thread until I'm sure I'm in the right area asking for help. For starters, the above way is how i've set up all my tests for the connections and until I've used an enum, hasn't given me any issues. 

If this is in fact a pgx issue I'm having, I'll gladly add more supporting information upon request.

Thank you

**Comment by jackc** [comment, marked as answer]
The error message is from the PostgreSQL server, so I suspect the issue is not directly from pgx.

enums can have some odd behavior connected to transactions, but I'm not sure if that is related to what's happening here or not.

**Reply by csphilli** [reply]
Thanks your for your help. I think i'll instead implement a different solution to my problem instead of enums.

I appreciate all the work you've done, thank you!
